### PR TITLE
fix: fix the construction of List<Null>

### PR DESCRIPTION
**Pull request by reswqa**
IIRC, This should be fixed as we discussed in discord.

### CODE REVIEW

File: crates/polars-core/src/series/implementations/null.rs [hunk 1]
```diff
@@ -43,7 +43,19 @@ impl PrivateSeriesNumeric for NullChunked {}
 
 impl PrivateSeries for NullChunked {
     fn compute_len(&mut self) {
-        // no-op
+        // NullChunked only has one chunk.
+        self.length = self.chunks[0].len() as IdxSize;
```
Comment (reswqa) [inline crates/polars-core/src/series/implementations/null.rs:47]: `amortized_iter_with_name` will change the chunk directly during iteration, but the length will not be re-compute before.

Comment (ritchie46) [reply crates/polars-core/src/series/implementations/null.rs:47]: I think we should sum the length of all chunks. We also should convert via `IdxSize::from` and remove the asserts below (they are wrong). See the `compute_len` on `ChunkedArray<T>`.

**Review by ritchie46 — COMMENTED** [review]
Thanks for the fix. I have left some comments.

File: crates/polars-core/src/series/implementations/null.rs (hunk 1 above)
Comment (reswqa) [reply crates/polars-core/src/series/implementations/null.rs]: > I think we should sum the length of all chunks.

Ah, I'm not realize that we had already made null chunking behavior equal to other dtypes before.

> We also should convert via IdxSize::from and remove the asserts below (they are wrong).

Make sense, seems that we just fixed this a few days ago for `ChunkedArray<T>`.

Updated this.

**Pull request merged**